### PR TITLE
+ core: TraversableOnce is Effector

### PR DESCRIPTION
**Pull request by Governa**
Replaces the old `List is Effector`

### CODE REVIEW

File: macroid-core/src/main/scala/macroid/util/Effector.scala [hunk 1]
```diff
@@ -9,8 +9,9 @@ trait Effector[F[_]] {
 }
 
 object Effector {
-  implicit object `List is Effector` extends Effector[List] {
-    def foreach[A](fa: List[A])(f: A ⇒ Any) = fa.foreach(f)
+
+  implicit def `TraversableOnce is Effector`[T[B] <: TraversableOnce[B]]: Effector[T] = new Effector[T] {
```
Comment (stanch) [inline macroid-core/src/main/scala/macroid/util/Effector.scala]: What do you think about making `Effector` contravariant instead? I’ve tried this and it seemed to work for List, Vector, Seq, etc:
```scala
trait Effector[-F[_]] {
  def foreach[A](fa: F[A])(f: A ⇒ Any): Unit
}

implicit object `TraversableOnce is Effector` extends Effector[TraversableOnce] {
  override def foreach[A](fa: TraversableOnce[A])(f: A ⇒ Any): Unit = fa.foreach(f)
}
```

**Comment by stanch** [comment]
Thanks! See my comment inline. If you want, I can make that modification on top of your commit myself.

**Comment by Governa** [comment]
Yeah... yours is much better. It doesn`t involve creating new objects for each type used.

The funny thing is that I was sure that I had tried it and it didn`t work. But it does work...

**Comment by stanch** [comment]
Hey, why would you close the PR? I can merge your commit if you amend it with that version :)

**Comment by stanch** [comment]
:+1:

**Pull request merged**